app/javascript/components/visualization/controls/DifferentialExpressionGroupPicker.jsx

```diff
@@ -76,77 +80,174 @@ function getMatchingDeOption(deObjects, group, clusterName, annotation) {
+      return deOption !== undefined && deOption !== newGroup
+    })
+    let groupHasRest = false
+    if (hasOneVsRestDe) {
+      groupHasRest = getMatchingDeOption(deObjects, newGroup, clusterName, annotation)
+      if (groupHasRest) {
+        newGroupsB.unshift('rest')
+      }
+    }
+    setDeGroupsB(newGroupsB)
+
+    if (newGroup === deGroupB || deGroupB && deGroupB === 'rest' && !groupHasRest) {
+      setDeGroupB(null) // Clear group B upon changing group A, if A === B
+      setDeGenes(null)
+      return
+    }
+
+    if (deGroupB) {
+      updateTable(newGroup, deGroupB)
+    }
+  }
+
+  /** Update group in differential expression picker */
+  async function updateDeGroupB(newGroup) {
+    setDeGroupB(newGroup)
+
+    updateTable(deGroup, newGroup)
   }
 
   return (
     <>
-      {!deGenes &&
-        <div className="flexbox-align-center flexbox-column">
-          <span>Compare one group to the rest</span>
+      <div className="differential-expression-picker">
+        {!deGenes && <p>Compare one group to another.</p>}
+        <div className="pairwise-select">
           <Select
-            defaultMenuIsOpen
+            defaultMenuIsOpen={!deGenes}
             options={getSimpleOptions(groups)}
-            data-analytics-name="de-group-select"
+            data-analytics-name="de-group-select-a"
+            className="differential-expression-pairwise de-group-select"
             value={{
               label: deGroup === null ? noneSelected : deGroup,
               value: deGroup
             }}
-            onChange={newGroup => updateDeGroup(newGroup.value)}
+            onChange={newGroup => updateDeGroupA(newGroup.value)}
             styles={clusterSelectStyle}
           />
         </div>
-      }
-      {deGenes &&
-      <div className="differential-expression-picker">
-        <div className="narrow-select">
+        <span className="vs-note">vs. </span>
+        <div className="pairwise-select pairwise-select-b">
           <Select
-            options={getSimpleOptions(groups)}
-            data-analytics-name="de-group-select"
+            options={getSimpleOptions(deGroupsB)}
+            data-analytics-name="de-group-select-b"
+            className="differential-expression-pairwise de-group-select"
             value={{
-              label: deGroup === null ? noneSelected : deGroup,
-              value: deGroup
+              label: !deGroupB ? noneSelected : deGroupB,
+              value: deGroupB
             }}
-            onChange={newGroup => updateDeGroup(newGroup.value)}
+            onChange={newGroup => updateDeGroupB(newGroup.value)}
             styles={clusterSelectStyle}
           />
         </div>
-        <span className="vs-note">vs. rest</span>
-        <br/>
-        <br/>
       </div>
-      }
+      {deGenes && <><br/><br/></>}
+    </>
+  )
+}
+
+/** Pick groups of cells for one-vs-rest-only differential expression (DE) */
+export function DifferentialExpressionGroupPicker({
```

Comment:
Agreed, I'll update this to something along those lines in my forthcoming PR for the author DE table, and link to that change from this thread when done.  (I merged just now before changing since it helps faster unblock Jon.)

Reply:
[Done](https://github.com/broadinstitute/single_cell_portal_core/pull/1813/commits/52974b81b5e9dceae3f3be3e6ed881844b1d012c)!  I went with "OneVsRestDifferentialExpressionGroupPicker" -- longer, but clearer and more guessable.